### PR TITLE
Add Parquet support to BigQuery

### PR DESCRIPTION
**Pull request by quartzmo**
Support loading the [Parquet](https://parquet.apache.org/) format, but not for extract jobs or as external datasource.

Note: There is currently a skipped acceptance test with the message skip "A GCS Parquet file is required to implement this test." The acceptance test using a local Parquet file is passing..

### CODE REVIEW

**Comment by blowmage** [comment]
LGTM. Is this PR ready to be accepted or are you still working on the items listed in the description such as uploading the .parquet file to a GCS bucket?

**Comment by quartzmo** [comment]
@blowmage I asked last week if there are any plans to provide a Parquet data file on GCS similar to `gs://cloud-samples-data/bigquery/us-states/us-states.avro`, and the answer seems to be that no one knows. We shouldn't wait for it.

**Comment by tswast** [comment]
@quartzmo What do you mean no one knows? `gs://cloud-samples-data/bigquery/us-states/us-states.parquet` is already used in the Python sample for this.

**Comment by quartzmo** [comment]
@tswast Thanks, I'll update the PR.

**Comment by quartzmo** [comment]
@tswast Updated and ready for review.

File: google-cloud-bigquery/acceptance/bigquery/dataset_test.rb [hunk 1]
```diff
@@ -292,6 +296,33 @@
...
+  end
+
+  it "imports data from GCS Parquet file and creates a new table with load" do
+    result = dataset.load(
+        table_parquet_id,
+        "gs://cloud-samples-data/bigquery/us-states/us-states.parquet")
+    result.must_equal true
+  end
+
+  it "imports data from GCS Parquet file and creates a new table with encryption with load" do
```
Comment (tswast) [inline google-cloud-bigquery/acceptance/bigquery/dataset_test.rb]: You can probably drop this test if you want. The fact that the other encryption test is Avro is more a side-effect that Avro is both fast to load and has an embedded schema.

Comment (quartzmo) [reply google-cloud-bigquery/acceptance/bigquery/dataset_test.rb]: ok, will do.

**Pull request merged**